inst/@sym/transpose.m

```diff
@@ -66,8 +66,11 @@
   end
 
   cmd = { 'x = _ins[0]'
-          'if x.is_Matrix:'
+          'if isinstance(x, MatrixBase):'
           '    return x.T'
+          'elif isinstance(x, NDimArray):'
+          '    xx = x.tolist()'
+          '    return Array(list(zip(*xx)))'
```

Comment:
should these go through your new helper function?  (not saying they should, just asking)

Reply:
Is it alright if I do

```
If isinstance(x, (MatrixBase, NDimArray)):
    xx = x.tolist()
    return make_matrix_or_array(list(zip(*xx)))
```

directly instead of splitting the case?

Reply:
This actually doesn't work because of issues with empty matrices. We have to split it into 2 cases.

Reply:
Replying to your original Q:

I thought about this but don't have a clear ans. Using `make_matrix_or_array` means sticking to a more consistent interface, while using `Array` is better for performance because no Expr checks are performed. I guess sticking to a more consistent interface is more important (?)

Reply:
As per my latest comment in #1124, we may want to use `is_2d_sym` and `is_non_matrix_2d_sym` for generality sake.

Reply:
Yeah that sounds reasonable.   A new PR perhaps